### PR TITLE
Adding command line useability improvements.

### PR DESCRIPTION
**Pull request by SasaKaranovic**
# Added

- Added `requirements.txt` to allow end users to install all dependencies using `pip install -r requirements.txt`
- Added checking if input file is correct type (.stl/.gcode)
- Added check for output folder and if it does not exist and attempt to create it before running transformation
- Added printout of the transformation 'settings' before running
- Added option to recognize transform options (angle, direction) from the .gcode file name
- Added help section. Each script can be run with `-h` to list available options and their description

# Modified
- Modified script to use `argparse` and accept command line arguments
- Script now accepts command line arguments to specify input file, output file, direction, angle etc.

# How to install script dependencies
Run `pip install -r requirements.txt` (assuming you have Python and PIP installed)

# How to run the script
## Transform
Transform `input.stl` STL file and save output to `transformed` folder. Use angle of 16 degrees and run 1 iteration. Default is `outward` transformation.

Example: 
`Transformation_STL_var_angle.py --file input.stl --output transformed --angle 16 --iterations 1` 

## Back transform GCode
Back transform `test_transformed_outward_16deg_0.8n_0.55mm_PETG_MK3S_32m.gcode` and save to `transformed` folder. Use filename to detect transformation parameters.

Example: 
`Transformation_STL_var_angle.py --file test_transformed_outward_16deg_0.8n_0.55mm_PETG_MK3S_32m.gcode --output transformed` 

Alternatively you could also specify angle and direction via command line parameters.

### CODE REVIEW

**Comment by zealws** [comment]
With some extra effort, the script could be modified to operate in a way compatible with [PrusaSlicer's post-processing script feature](https://help.prusa3d.com/article/post-processing-scripts_283913). That would make the second step of post-processing the gcode way less tedious, since you could prepare a slicer profile that does that step automatically.

We would need to modify the script to accept the filename as a single parameter and modify the filename in place, which is probably a matter of making that the default behavior if no other options are provided.

File: Scripts for Variable Angle/Backtransformation_GCode_var_angle.py [hunk 1]
```diff
@@ -1,18 +1,8 @@
 import re
```
Comment (zealws) [inline Scripts for Variable Angle/Backtransformation_GCode_var_angle.py:1]: I don't see the change in the PR, but you probably want to make both scripts executable.

On linux or OSX:
```
chmod +x 'Scripts for Variable Angle/Backtransformation_GCode_var_angle.py'
```

No clue how to do that on windows :shrug:

File: Scripts for Variable Angle/Transformation_STL_var_angle.py [hunk 2]
```diff
@@ -1,18 +1,8 @@
 import numpy as np
```
Comment (zealws) [inline Scripts for Variable Angle/Transformation_STL_var_angle.py:1]: Same here

```
chmod +x 'Scripts for Variable Angle/Transformation_STL_var_angle.py'
```

File: Scripts for Variable Angle/Backtransformation_GCode_var_angle.py (hunk 1 above)
Comment (zealws) [reply Scripts for Variable Angle/Backtransformation_GCode_var_angle.py:1]: Oh, also need a shebang for unix systems
```suggestion
#!/usr/bin/env python
import re
```

Sorry for leaving each comment one-at-a-time. I'm working through it right now.

Comment (FacuM) [reply Scripts for Variable Angle/Backtransformation_GCode_var_angle.py:1]: > No clue how to do that on windows 🤷

Windows simply doesn't care about that, if the file's extension is `.py` it'll try to run it regardless of its contents and/or permissions.

**Pull request merged**